### PR TITLE
chore: Bump codemirror

### PR DESCRIPTION
**Pull request by rschristian**
Bumps CodeMirror up to v6, somewhat pulled from #793 (man was that PR a gold mine of stuff). Bit of an increase on the bundle size (loading the 'Simple Counter' w/ no cache: `393kb` -> `448kb`, `-12kb` for the uncompressed favicon in preview, or ~43kb bump), but I don't think it's too bad.

I've enabled auto completion (basic, but not bad) and bracket matching with this bump, seem like general nice-to-haves and I am already quite happy with these. There seems to be quite a bit more we can also enable/play around with in the future too.

### CODE REVIEW

File: src/components/controllers/tutorial/index.jsx [hunk 1]
```diff
@@ -134,15 +128,14 @@ export class Tutorial extends Component {
 		this.setState({ error: null });
 	};
 
-	render({ html, meta, loading }, { code, error }) {
+	render({ html, meta, loading: routerLoading }, { code, error, loading: tutorialLoading }) {
```
Comment (rschristian) [inline src/components/controllers/tutorial/index.jsx:131]: Drive-by loading mechanism improvement over #1143 

Still needs work, but incorporates both routing (markdown and tutorial bundles) with the tutorial orchestration (editor & repl runner) better.

File: src/components/controllers/repl/index.jsx [hunk 2]
```diff
@@ -111,26 +113,17 @@ export class Repl extends Component {
 	}
 
 	componentDidMount() {
-		Promise.all([
-			import('../../code-editor'),
-			import('./runner')
-		]).then(([CodeEditor, Runner]) => {
```
Comment (rschristian) [inline src/components/controllers/repl/index.jsx:117]: No real need to lazily init like this, Vite is smart enough to combine the code editor & runner imports so that tutorial & repl can share them, which is nice.

Might need to keep an eye on that though. Might also revert yet, dunno. Thought I'd try it anyhow.

File: src/components/code-editor/index.jsx [hunk 3]
```diff
@@ -1,99 +1,72 @@
...
-	componentWillUnmount() {
-		let wrapper = this.editor && this.editor.getWrapperElement();
-		if (wrapper) this.base.removeChild(wrapper);
-		this.editor = null;
-	}
+		return () => {
+			view.destroy();
+			setEditor(null);
+		};
+	}, [props.baseExampleSlug]);
```
Comment (rschristian) [inline src/components/code-editor/index.jsx:69]: TODO: This probably needs some work.

File: src/components/code-editor/index.jsx [hunk 4]
```diff
@@ -1,99 +1,72 @@
...
-		if (this.errors) {
-			this.editor.operation(() => {
-				this.errors.forEach(e => e.clear());
-			});
-			this.errors.length = 0;
-		}
-
-		if (!error || !error.loc) return;
+// Custom theme that better matches our Prism config, though
+// the lexer is somewhat limited so it still deviates
```
Comment (rschristian) [inline src/components/code-editor/index.jsx:16]: Theme definitely isn't perfect, but this is a bit closer to our Prism config than the previous set up was.

File: src/components/controllers/repl/index.jsx [hunk 5]
```diff
@@ -308,15 +306,6 @@ export class Repl extends Component {
...
 		<loading-bar showing={!!loading} />
-		<style>{`
-			main {
-				height: 100% !important;
-				overflow: hidden !important;
-			}
-			footer {
-				display: none !important;
-			}
-		`}</style>
```
Comment (rschristian) [inline src/components/controllers/repl/index.jsx:319]: Drive-by change, but I see no reason why this needs to be a style tag?

File: src/components/code-editor/index.jsx [hunk 6]
```diff
@@ -1,99 +1,106 @@
...
+				bracketMatching(),
+				autocompletion(),
+				javascript({ jsx: true }),
+				keymap.of([indentWithTab, ...defaultKeymap, ...historyKeymap]),
+				[theme, syntaxHighlighting(highlightStyle, { fallback: true })],
+				EditorView.updateListener.of(update => {
+					// Ignores changes from swapping out the editor code programmatically
+					if (isViewUpdateFromUserInput(update)) {
+						props.onInput(update.state.doc.toString());
+					}
```
Comment (rschristian) [inline src/components/code-editor/index.jsx:78]: TODO: Upon further testing, this seems to be a race condition. Any input triggers two events, one in which `isViewUpdateFromUserInput()` returns `true`, the second, `false`. Which has the updated doc string is inconsistent.

File: src/components/controllers/repl-page.jsx [hunk 7]
```diff
@@ -38,42 +38,31 @@ export default function ReplPage() {
  *
  * ?code -> ?example -> localStorage -> simple counter example
  */
-function initialCode(query) {
-	let code, slug;
```
Comment (rschristian) [inline src/components/controllers/repl-page.jsx:42]: TODO: Split this out. Needs to land still, reduces the mess, but not necessarily needed here.

File: src/components/code-editor/index.jsx [hunk 8]
```diff
@@ -1,99 +1,106 @@
...
-					<div ref={r => (ref = r)} class={style.lintError}>
-						<pre style={`padding-left:${left}px;`}>▲</pre>
-						<div>🔥 {error.message.split('\n')[0]}</div>
-					</div>
-				);
-				render(errorLine, this.scratch);
-				this.errors = [this.editor.addLineWidget(error.loc.line - 1, ref)];
-			});
-		}, 1000);
-	}
```
[the remark this answers is not in the document]
Comment (rschristian) [reply src/components/code-editor/index.jsx:41]: Actually, the tutorial does seem to show errors. Will need to take another look.

[the remark this answers is not in the document]
Comment (rschristian) [reply src/components/code-editor/index.jsx:41]: The error positioning is pretty touchy & limited, we can't effectively map the built output back to the code editor input.

Redoing the compilation is a can of worms I don't want to address here. For now, I've disabled the error overlay in the editor as it barely works and tends to point at the wrong line entirely.

**Review by JoviDeCroock — APPROVED** [review]
